testcases/Lcov.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python3
+# IBM_PROLOG_BEGIN_TAG
+# This is an automatically generated prolog.
+#
+# $Source: op-test-framework/testcases/Lcov.py
+#
+# OpenPOWER Automated Test Project
+#
+# Contributors Listed Below - COPYRIGHT 2023
+# [+] International Business Machines Corp.
+#
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+# IBM_PROLOG_END_TAG
+
+'''
+Lcov setup Testcase
+--------------------
+
+This test case to setup Lcov 
+'''
+
+import unittest
+import os
+import time
+import tempfile
+import OpTestConfiguration
+from common.OpTestUtil import OpTestUtil
+from common.OpTestSystem import OpSystemState
+
+import logging
+import OpTestLogger
+log = OpTestLogger.optest_logger_glob.get_logger(__name__)
+
+
+class LcovSetup(unittest.TestCase):
+    '''
+    Gather the gcov ran data using lcov tool
+    '''
+
+    def setUp(self):
+        '''
+        get the configuration details
+        '''
+        conf = OpTestConfiguration.conf
+        self.cv_HOST = conf.host()
+        self.cv_IPMI = conf.ipmi()
+        self.cv_SYSTEM = conf.system()
+        self.util = OpTestUtil(OpTestConfiguration.conf)
+        self.c = self.cv_SYSTEM.cv_HOST.get_ssh_connection()
+        self.distro = self.util.distro_name()
+        self.k_version = "".join(self.cv_HOST.host_run_command("uname -r"))
+
+    def runTest(self):
+        '''
+        Test for installing the lcov
+        '''
+        log.info("distro_name=%s" %self.distro)
+        if self.distro == 'rhel':
+            cmd = "yum install"
+        elif self.distro == 'sles':
+            cmd = "zypper install"
+        dep_packages = ['perl*', 'tiny*']
+        log.info("installing the dependency packages")
+        for pkg in dep_packages:
+            self.c.run_command(cmd + " " + pkg + " -y")
+        time.sleep(5)
+        log.info("changing dir to /home")
+        self.c.run_command("cd /home/")
+        url = 'git clone https://github.com/linux-test-project/lcov.git'
+        self.c.run_command(url)
+        log.info("changing dir to /lcov")
+        self.c.run_command("cd lcov")
+        self.c.run_command("pwd")
+        self.c.run_command("make install")
+
+
+class LcovCounterZero(LcovSetup):
+    '''
+    Reset the lcov counter to zero before data gather
+    '''
+    def runTest(self):
+        '''
+        Running the test for Zeroing the lcov
+        '''
+        self.c = self.cv_SYSTEM.cv_HOST.get_ssh_connection()
+        log.info("Zeroing the lcov")
+        self.c.run_command("echo 1 > /sys/kernel/debug/gcov/reset")
+        self.c.run_command("lcov --zerocounters")
```

Comment:
can we have this as  a class method instead of separate class ?

Reply:
reset counter can go in LcovSetup()  itself as a method .. any need to have it as a separate class ?

Reply:
This is as per the plan that we have discussed long back to have two different files.

Reply:
ok